bindings/pyroot/pythonizations/python/ROOT/pythonization/_roofit/_rooargset.py

```diff
@@ -17,6 +17,20 @@
 
 
 class RooArgSet(RooAbsCollection):
+    def __init__(self, *args, **kwargs):
+        """Pythonization of RooArgSet constructor to support implicit
+        conversion from Python sets.
+        """
+        # Note: This simple Pythonization caused me days of headache.
+        # Initially, I was also checking of `len(kwargs) == 0`, but it just
+        # didn't work. Eventually, I understood that when cppy attempts
+        # implicit conversion, a magic `__cppyy_no_implicit=True` keyword
+        # argument is added, hence the `len(kwargs) == 0` check breaks the
+        # implicit conversion!
+        if len(args) == 1 and isinstance(args[0], set):
+            return self.__init__(*args[0], **kwargs)
```

Comment:
Shouldn't this invoke `self._init` ? Otherwise you just invoke again `__init__` with the unpacked set to in the end invoke `_init`.

Reply:
Yes that's true, we can save one recursion call here. I will updated the PR.